src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleCache.java

```diff
@@ -0,0 +1,86 @@
+package net.sf.jabref.logic.citationstyle;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.sf.jabref.BibDatabaseContext;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.event.EntryChangedEvent;
+import net.sf.jabref.model.event.EntryRemovedEvent;
+
+import com.google.common.eventbus.Subscribe;
+
+
+/**
+ * Caches the generated Citations for quicker access
+ */
+public class CitationStyleCache {
```

Comment:
Is generating the preview so slow that a cache is needed? You are investing quite a bit of memory to save what looks to me like a few stringy operations.